library/std/src/sys/pal/windows/stdio.rs

```diff
@@ -84,21 +84,43 @@ fn is_console(handle: c::HANDLE) -> bool {
     unsafe { c::GetConsoleMode(handle, &mut mode) != 0 }
 }
 
+/// Returns true if the attached console's code page is currently UTF-8.
+#[cfg(not(target_vendor = "win7"))]
+fn is_utf8_console() -> bool {
+    unsafe { c::GetConsoleOutputCP() == c::CP_UTF8 }
```

Comment:
Is this cheap enough to call on every write? I guess we're already calling GetConsoleMode on every write...

Reply:
Yeah, I don't love it but it is cheap. Especially when compared to the previous code where if `is_console` is true then it will be doing a full UTF-8 to UTF-16 conversion using a small buffer (necessitating multiple calls to `write`).